src/features/wallet/AddPastelPromoCodeModal.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FormEvent, useState, useEffect } from 'react'
+
+import { importPastelPromoCode } from 'common/utils/PastelPromoCode'
+import { TitleModal } from 'common/components/Modal'
+import { Input } from 'common/components/Inputs'
+import { Button } from 'common/components/Buttons'
+import { isValidPrivateKey } from 'common/utils/wallet'
+import { useCurrencyName } from 'common/hooks/appInfo'
+import { WalletRPC } from 'api/pastel-rpc'
+import { formatNumber } from 'common/utils/format'
+import congratulations from 'common/assets/icons/ico-congratulations.svg'
+
+type TAddPastelPromoCodeModalProps = {
+  isOpen: boolean
+  handleClose: () => void
+  fetchData: () => void
+}
+
+export default function AddPastelPromoCodeModal({
+  isOpen,
+  handleClose,
+  fetchData,
+}: TAddPastelPromoCodeModalProps): JSX.Element {
+  const walletRPC = new WalletRPC()
+  const currencyName = useCurrencyName()
+  const [isValidPromoCode, setValidPromoCode] = useState<boolean>(false)
+  const [isLoading, setLoading] = useState<boolean>(false)
+  const [message, setMessage] = useState<string>('')
+  const [pastelPromoCode, setPastelPromoCode] = useState<string>('')
+
+  useEffect(() => {
+    if (isOpen) {
+      setValidPromoCode(false)
+      setMessage('')
+      setLoading(false)
+    }
+  }, [isOpen])
+
+  const handleAddPromoCode = async () => {
+    setMessage('')
+    if (isValidPrivateKey(pastelPromoCode)) {
+      setLoading(true)
+      const result = await importPastelPromoCode(pastelPromoCode)
+      if (result) {
+        const addresses = await walletRPC.fetchTandZAddresses()
+        const promoCodeBalance = addresses.filter(
+          address => address.address === result,
+        )
+        setValidPromoCode(true)
+        setMessage(
+          formatNumber(parseFloat(promoCodeBalance[0]?.amount.toString()) || 0),
```

Comment:
If no address from promo code user will see NaN.
` || 0` this code in the end of line won't be applied because NaN is not falsey